src/internal/form.ts

```diff
@@ -72,9 +72,8 @@ export class FormControlController implements ReactiveController {
         const formId = input.form;
```

Comment:
Thanks for looking into this.

However if both the form and the element are detached, the element is still connected to the form:
```js
  const frm = document.createElement("form")
  const btn = document.createElement("button")
  frm.appendChild(btn)
  console.log(btn.isConnected) // false
  console.log(btn.form) // <form><button></button></form>
```

I can't really assess whether this case is problematic and how to handle it best.
But your change seems reasonable to me so I'm onboard integrating it in case my hints don't change anything for you.

Reply:
Ahhh interesting. Good catch. I think it should be good to go as is then.